pkg/utils/grpc.go

```diff
@@ -19,11 +22,36 @@ func ParseEndpoint(ep string) (string, string, error) {
 	return "", "", fmt.Errorf("invalid endpoint: %v", ep)
 }
 
-func LogGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-	klog.V(5).Infof("GRPC call: %s", info.FullMethod)
-	resp, err := handler(ctx, req)
-	if err != nil {
-		klog.ErrorS(err, "GRPC request error")
+// UnaryServerInterceptor provides metrics around Unary RPCs.
+func UnaryServerInterceptor() func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		start := time.Now()
+		reporter := metrics.NewStatsReporter()
+
+		var err error
+		defer func() {
+			if err != nil {
+				reporter.ReportRequest(ctx, fmt.Sprintf("%s_%s", metrics.GrpcOperationTypeValue, getGRPCMethodName(info.FullMethod)), metrics.ErrorStatusTypeValue, time.Since(start).Seconds(), err.Error())
+				return
+			}
+			reporter.ReportRequest(ctx, fmt.Sprintf("%s_%s", metrics.GrpcOperationTypeValue, getGRPCMethodName(info.FullMethod)), metrics.SuccessStatusTypeValue, time.Since(start).Seconds())
```

Comment:
same comment as `ReportRequest` for encrypt/decrypt

Reply:
refactored.